packages/core/src/types.ts

```diff
@@ -75,6 +73,381 @@ export interface PositionWithOffset extends Position {
  */
 export type SourceRange = [number, number];
 
+//------------------------------------------------------------------------------
+// Rules
+//------------------------------------------------------------------------------
+
+/**
+ * What the rule is responsible for finding:
+ * - `problem` means the rule has noticed a potential error.
+ * - `suggestion` means the rule suggests an alternate or better approach.
+ * - `layout` means the rule is looking at spacing, indentation, etc.
+ */
+export type RuleType = "problem" | "suggestion" | "layout";
+
+/**
+ * The type of fix the rule can provide:
+ * - `code` means the rule can fix syntax.
+ * - `whitespace` means the rule can fix spacing and indentation.
+ */
+export type RuleFixType = "code" | "whitespace";
+
+/* eslint-disable @typescript-eslint/consistent-indexed-object-style -- Needs to be interface so people can extend it. */
```

Comment:
Aside: I hit this myself a few times recently. It's really annoying 😕. https://github.com/typescript-eslint/typescript-eslint/issues/7863 - added to my "todo soon" if nobody else gets to it...